packages/core/src/agent/MessageSender.ts

```diff
@@ -167,6 +196,11 @@ export class MessageSender {
     // Retrieve DIDComm services
     const allServices = this.transportService.findDidCommServices(connection)
     const reachableServices = allServices.filter((s) => !isDidCommTransportQueue(s.serviceEndpoint))
+    if (options && options.preferredTransport) {
```

Comment:
I see you use sort here, but filter in another method. I think sort is the way to go. Could you extract the sorting logic into a separate method. Also I see you use the protocol scheme as the preferred transport indicator. Maybe we should rename `preferredTransport` to `preferredProtocolScheme` to better indicate what the value of `preferredTransport` is?

Reply:
```suggestion
    if (options?.preferredTransport) {
```